libraries/HTTPClient/src/HTTPClient.cpp

```diff
@@ -261,6 +261,10 @@ bool HTTPClient::beginInternal(String url, const char* expectedProtocol)
     url.remove(0, (index + 3)); // remove http:// or https://
 
     index = url.indexOf('/');
+    if (index == -1) {
+        log_e("the path of url must not be empty");
+        return false;
```

Comment:
I agree a check is needed here to prevent using `-1` for `index`, but I don't think it should be an error. If you type in "example.com" into a browser it defaults the path to `/` for you. Likely this code should do the same in this case. Perhaps something like this inside this new condition block:
```
  url.append("/");
  index = url.indexOf('/');
```
It will incur the cost of calling indexOf a second time but I don't think there are many options around this.

Reply:
I think so too. I'll fix this.
> I don't think there are many options around this.  

The method `length` can be also used.